src/main/java/Duke.java

```diff
@@ -1,10 +1,368 @@
+                String description = task.substring(8, task.length());
+                tasks[counter] = new Todo(description, "[T]");
+                if (completion_status.equals("[\u2713]")) {
+                    tasks[counter].updateisDone(true);
+                }
+                counter++;
+            }
+            else if (task_type.equals("[D]")) { // it's a deadline
+                String task_description = task.substring(8, task.indexOf("("));
+
+                String[] split_str = task.split("by:");
+                String deadline = split_str[1].replace(")", ""); // this is the deadline which you should be doing date recognition
+
+                tasks[counter] = new Deadline(task_description, "[D]", deadline);
+                if (completion_status.equals("[\u2713]")) {
+                    tasks[counter].updateisDone(true);
+                }
+                counter++;
+            }
+            else { // it's an event already
+                String task_description = task.substring(8, task.indexOf("("));
+
+                String[] split_str = task.split("at:");
+                String deadline = split_str[1].replace(")", ""); // this is the deadline which you should be doing date recognition
+
+                tasks[counter] = new Event(task_description, "[E]", deadline);
+                if (completion_status.equals("[\u2713]")) {
+                    tasks[counter].updateisDone(true);
+                }
+                counter++;
+            }
+        }
+    }
+
+    public static void findTask(String keyword) {
+        //Task[] potential_tasks = new Task[100];
+        int potential_task_counter = 0;
+        for (int i = 0; i < counter; i++) {
+            if (tasks[i].getDescription().contains(keyword)) {
+                potential_task_counter++;
+            }
+        }
+        System.out.println("Here are the matching tasks in your list: ");
+        for (int i = 0; i < counter; i++) {
+            if (tasks[i].getDescription().contains(keyword)) {
+                System.out.println(Integer.toString(i+1) + ". " + tasks[i].getTaskType() + tasks[i].getCompletionStatus() + tasks[i].getDescription() + tasks[i].getDeadline());
+            }
+        }
+    }
+
+    public static void addTodo(String description) throws DukeException {
+        if (description.equals("")) {
+            throw new DukeException("\u2639 OOPS!!! The description of a todo cannot be empty.");
+        }
+        tasks[counter] = new Todo(description, "[T]");
+        Task current_task = tasks[counter];
+        counter++;
+
+        System.out.println("Got it. I've added this task:");
+        System.out.println(current_task.getTaskType() + current_task.getCompletionStatus()
+                + current_task.getDescription());
+        System.out.println("Now you have " + Integer.toString(counter) + " tasks in the list.");
+    }
+
+    public static void addDeadline(String description, String deadline) {
```

Comment:
Consider having another class to handle command behaviours as stated in the More-OOP increment in week 3.